### PR TITLE
Prepare for DIAproteomics Release 1.2.3 (Mt. Elbrus South Route)

### PR DESCRIPTION
**Pull request by Leon-Bichmann**
The PR includes the following changes:

 - updates of DIAproteomics to the latest nf-core template 1.13.3
 - introduction of an optional mztab-like output file
 - introduction of an optional column in the input sample sheet (MSstats_BioReplicate) that can be used in the statistical post-processing if available

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
 - [x] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
 - [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/diaproteomics/tree/master/.github/CONTRIBUTING.md)
 - [x] If necessary, also make a PR on the nf-core/diaproteomics _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint .`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: bin/mztab_output.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+    #parse library and rename columns
+    df_lib=pd.read_csv(lib,sep='\t')
+    df_lib=df_lib[['ModifiedPeptideSequence', 'PrecursorMz', 'ProteinId']]
+    df_lib.columns=['PeptideSequence', 'PrecursorMz', 'ProteinName']
+
+    #parse experimental design and rename columns
+    df_I=pd.read_csv(exp_design, sep='\t')
+    if any(['/' in f for f in df_I['Spectra_Filepath'].values.tolist()]):
+       files=[f.replace('.mzML','').replace('.raw','').replace('.Raw','').replace('.RAW','').split('/')[-1] for f in df_I['Spectra_Filepath'].values.tolist()]
```
Comment (timosachsenberg) [inline bin/mztab_output.py]: is "mzml" encoutnered in the wild?

File: bin/mztab_output.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+    cols=df_prot.columns.tolist()
+    df_prot['PRH']='PRT'
+    df_prot=df_prot[['PRH']+cols]
+    df_prot['num_peptides']=num_peptides
+    df_prot['num_peptides_distinct']=num_peptides_distinct
+    for c in remaining_prh_cols:
+        df_prot[c]='null'
+
+    df_prot.to_csv('PRT_dataframe.csv',index=False, sep='\t')
```
Comment (timosachsenberg) [inline bin/mztab_output.py:211]: why write out and read back in again?

File: bin/mztab_output.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    df_pep['PEH']='PEP'
+    df_pep=df_pep[['PEH']+cols]
+    df_pep['retention_time']=retention_time
+    df_pep['search_engine']='OpenSwathWorkflow'
+    for c in remaining_pep_cols:
+        df_pep[c]='null'
+
+    df_pep.to_csv('PEP_dataframe.csv',index=False, sep='\t')
+
+    op=open('PEP_dataframe.csv','r')
```
Comment (timosachsenberg) [inline bin/mztab_output.py:237]: same here

File: bin/mztab_output.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    opr_pep=op.readlines()
+    op.close()
+
+    #reformat MTD header
+    header=['\t'.join(['MTD','mzTab-version','1.0'])+'\n']
```
Comment (timosachsenberg) [inline bin/mztab_output.py:242]: I think there are some more mandatory meta values that you could easily add. See e.g. https://psidev.info/sites/default/files/2017-07/v_1_0_0_20minute_guide_mzTab.pdf
mzTab-mode etc. can be easily hard coded

File: nextflow_schema.json [hunk 5]
```diff
@@ -365,36 +389,44 @@
...
                     ],
-                    "description": "DIAlignR XICfilter parameter"
+                    "description": "DIAlignR XICfilter parameter",
+                    "fa_icon": "fas fa-funnel-dollar"
                 },
                 "DIAlignR_parallelization": {
                     "type": "boolean",
                     "default": true,
-                    "description": "Whether DIAlignR should be executed using multithreading (may cause errors)"
+                    "description": "Whether DIAlignR should be executed using multithreading (may cause errors)",
```
Comment (timosachsenberg) [inline nextflow_schema.json:398]: is this still the case for the current version?

**Review by timosachsenberg — APPROVED** [review]
I think the pseudo mzTab could be further extended. Not sure if critical.

**Comment by timosachsenberg** [comment]
do you have a test that covers the new BioReplicate column?

**Comment by Leon-Bichmann** [comment]
Thanks @timosachsenberg , ill have a look at this tomorrow

**Comment by Leon-Bichmann** [comment]
> do you have a test that covers the new BioReplicate column?

ok, so what I am trying now is to include the BioReplicate column as part of the full test, that will be run on AWS after the next release. I'll make it run locally, however, msstats did not work with the tiny data set that is run by the CI

**Comment by Leon-Bichmann** [comment]
> I think the pseudo mzTab could be further extended. Not sure if critical.

ok I'll add a few more metavalue entries

File: nextflow_schema.json (hunk 5 above)
Comment (Leon-Bichmann) [reply nextflow_schema.json:398]: Not sure, but maybe its good to have this option, in case there are problems

File: bin/mztab_output.py (hunk 4 above)
Comment (Leon-Bichmann) [reply bin/mztab_output.py:242]: I'll have another look what else would be easy to incorporate

File: bin/mztab_output.py (hunk 2 above)
Comment (Leon-Bichmann) [reply bin/mztab_output.py:211]: I admit is was laziness, I didn't find a function to write data frames of different dimensions below each other in a text file in a quick first search.

File: bin/mztab_output.py (hunk 1 above)
Comment (Leon-Bichmann) [reply bin/mztab_output.py]: True, will add it to the wilderniss

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,19 @@
...
+
+### `Added`
+
+Optional mzTab output
+Optional BioReplicate annotation for MSstats
+
+## v.1.2.3dev - [25.03.21]
+
+### `Added`
+
```
Comment (ewels) [inline CHANGELOG.md]:
```suggestion
```

**Review by ewels — APPROVED** [review]
Template update all looks great!

**Comment by ewels** [comment]
The CI tests are warning about some params that aren't in the schema:

```

WARN: Found unexpected parameters:
25
* --DIAlign-R_parallelization: true
26
* --DIAlign-R_XICfilter: sgolay
27
* --DIAlign-R_query_FDR: 1
28
* --DIAlign-R_align_FDR: 1
29
* --DIAlign-R_unalign_FDR: 1
30
* --DIAlign-R_analyte_FDR: 1
31
* --DIAlign-R_global_align_FDR: 1
```

I thought that the linting should pick this up too 🤔

**Comment by Leon-Bichmann** [comment]
> > do you have a test that covers the new BioReplicate column?
> 
> ok, so what I am trying now is to include the BioReplicate column as part of the full test, that will be run on AWS after the next release. I'll make it run locally, however, msstats did not work with the tiny data set that is run by the CI

My local BioReplicate Test worked

File: bin/mztab_output.py (hunk 4 above)
Comment (Leon-Bichmann) [reply bin/mztab_output.py:242]: done

File: bin/mztab_output.py (hunk 1 above)
Comment (Leon-Bichmann) [reply bin/mztab_output.py]: done

**Pull request merged**